pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -258,6 +289,10 @@ func (nm *NodeManager) removeNode(node *v1.Node) {
 	nm.nodeInfoLock.Lock()
 	delete(nm.nodeInfoMap, node.ObjectMeta.Name)
 	nm.nodeInfoLock.Unlock()
+
+	nm.zoneInfoLock.Lock()
+	delete(nm.zoneInfoMap, node.ObjectMeta.Name)
+	nm.zoneInfoLock.Unlock()
 }
```

Comment:
lets use `defer` if possible.

Reply:
Done.

Reply:
Still not fixed.

Reply:
My bad. Fixed the code to use defer pattern for all access to zoneInfoLock.